src/Modal/Modal.component.tsx

```diff
@@ -0,0 +1,137 @@
+    margin: ${({ oversized, margin }: ModalProps) =>
+        margin || (oversized ? '2rem' : 'auto')};
+
+    display: flex;
+    align-items: center;
+    flex-direction: column;
+    justify-content: space-between;
+
+    border-radius: ${sizes.border.radius.modal};
+    background: ${({ background = colors.white.base }: ModalProps) =>
+        background};
+    ${({ color }: ModalProps) =>
+        css`
+            color: ${color};
+        `}
+    box-shadow: ${({
+        shadow = '0 0.375rem 0.5rem 0.25rem rgba(0,0,0,0.13)',
+    }: ModalProps) => shadow};
+
+    overflow: hidden;
+
+    // The Modal Content component only gets its bottom padding if
+    // its the last-child of the modal. Furthermore, if the Modal Header
+    // comes before it, then the Content loses its top padding
+    // (see Modal Header styles). The goal is that we have full padding
+    // in isolation, but lose the top or bottom padding if the Header or
+    // Footer (respectively) are also used.
+    ${StyledContent} {
+        padding: ${({ size = defaultSize }: ModalProps) =>
+            `4rem ${Sizes[size].contentPadding}rem 0 ${
+                Sizes[size].contentPadding
+            }rem`};
+        &:last-child {
+            padding-bottom: ${({ size = defaultSize }: ModalProps) =>
+                `${Sizes[size].contentPadding}rem`};
+        }
+    }
+    ${StyledHeader} {
+        padding-left: ${({ size = defaultSize }: ModalProps) =>
+            Sizes[size].contentPadding}rem;
+    }
+    ${StyledFooter} {
+        min-height: ${({ size = defaultSize }: ModalProps) =>
+            Sizes[size].footerHeight}rem;
+    }
+`;
+
+interface ModalProps extends StyledReactModal.ModalProps {
+    size?: ModalSize;
+    background?: string;
+    oversized?: boolean;
+    color?: string;
+    width?: string;
+    height?: string;
+    margin?: string;
+    shadow?: string;
+    children?: any;
+    className?: string;
+}
+
+export const Modal = ({
+    children,
+    className,
+    oversized,
+    backgroundProps,
+    ...props
+}: ModalProps): React.ReactElement<ModalProps> => (
+    <StyledModal
+        {...{
+            className: classnames('anchor-modal', className),
+            oversized,
+            backgroundProps: { ...backgroundProps, oversized },
+            ...props,
+        }}
+    >
+        {children}
+    </StyledModal>
+);
+
+Modal.Content = ModalContent;
+Modal.Header = ModalHeader;
+Modal.Footer = ModalFooter;
+Modal.Close = ModalClose;
+
+interface BaseModalBackgroundProps {
+    opacity?: number;
+    oversized?: boolean;
+}
+
+const CustomModalBackground = styled(StyledReactModal.BaseModalBackground)`
+    ${({ oversized }: BaseModalBackgroundProps) =>
+        oversized &&
+        css`
+            align-items: initial;
+        `}
+    overflow-y: auto;
+    background-color: ${({ opacity = 0.6 }: BaseModalBackgroundProps) =>
+        opacify(opacity, 'rgba(0,0,0,0)')};
```

Comment:
How does this work? It's taking a zero valued alpha of black and then changing the opacity? Why not just `rgb(0, 0, 0)`?

Reply:
I believe the way `opacify` works it adds additional opacity to something already semi-transparent. It doesn't _set_ the opacity, so rgba(0,0,0) would always remain at opacity 1.

But why don't I just do `rgba(0,0,0,${opacity})`? Derp

Reply:
I've avoided the docs for polished as I'm on the fence about deprecating it, so I did not know that's how that method worked. Definitely like the leaner approach you outlined, though.